tests/gradients/test_parameter_shift.py

```diff
@@ -311,6 +311,10 @@ def test_grad_recipe_parameter_dependent(self):
         shift rule"""
 
         class RX(qml.RX):
+            @property
+            def parameter_frequencies(self):
+                raise qml.operation.OperatorPropertyUndefined

```

Comment:
gosh our new logic flow is too clever.

I suppose @dwierichs this is an example of where we would want to prioritize `grad_recipe` if defined?

Reply:
Yes, exactly :D Next PR! :)

Reply:
ticket created.